plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      "kube-" + s.SchedulerName,
```

Comment:
> +1 to leaving the default lock object the same (by whatever mechanism)

I think we'll deprecate kube-scheduler after several releases. We also need to document it.